README.md

```diff
@@ -2,4 +2,329 @@
+
+Arrays are ordered by index; sets hold unique values; dictionaries hold unique keys with values.
+
+**Objective-C:** *Swift's* `Array`, `Set`, *and* `Dictionary` *classes are interoperable with* `NSArray`, `NSSet`, *and* `NSDictionary`.
+
+### Generic Collections
+
+Since Swift is a strongly-typed language, its collection types are also "typed", meaning that they are given knowledge regarding their contents. Apple notes in the [*Collection Types* chapter](https://developer.apple.com/library/prerelease/ios/documentation/Swift/Conceptual/Swift_Programming_Language/CollectionTypes.html#//apple_ref/doc/uid/TP40014097-CH8-ID105) that its `Array`, `Set`, and `Dictionary` types "are implemented as *generic collections*."
+
+Uses of the keywords `T`, `Element`, and `Item` denote a *generic*. There is no meaningful distinction between the terms and you will see all three of them in various documents, both from Apple and from answers and blogs written by outside developers.
+
+We'll discuss generics in further detail in upcoming lessons. The effect of this is that the compiler gives individual instances of these collections knowledge of their membership types when they are declared, but otherwise the collection types themselves work exactly the same way whether they are told that they hold `Int`s, or `String`s, or anything else.
+
+### `let` versus `var`
+
+Collections declared using `let` are *immutable* and cannot be altered at all after they are initialized. If you wish to make a collection *mutable*, declare it with `var`. Apple recommends declaring collections with `let` as a default until you encounter the need to alter its contents, claiming that `let` allows the compiler to streamline performance if the collection never needs to be altered.
+
+### Pass-by-value
+
+Swift's collections types are passed by *value* and not by *reference*. When a collection instance is assigned to another collection instance, a copy of the instance is generated and assigned to the new instance. The effect of this is important to understand: **changes made to a collection after it is passed are not reflected in the state of the original collection.**
+
+This is true both for local instances and instances passed into functions and methods via arguments. *If you wish to preserve the mutated state of an instance that is passed-by-value, you have to assign the mutated version back to the original instance.*
+
+## `Array`
+
+### Type Annotation
+
+To explicitly declare the type of an empty array, wrap the type within square brackets `[` `]`. You can use either type annotation to set it equal to empty square brackets, or the initializer syntax that uses a parenthesis following the array type:
+
+```swift
+let array: [Element] = []      // type annotation
```

Comment:
Instead of "Element" I would say "Type". When I see element I kinda wanna put `let array: ["mark"]` because that's an element in the array